### PR TITLE
lvm2: cleanup & fix eval error when systemd=null

### PR DESCRIPTION
**Pull request by joachifm**

### CODE REVIEW

**Comment by joachifm** [comment]
@GrahamcOfBorg build lvm2

File: pkgs/os-specific/linux/lvm2/default.nix [hunk 1]
```diff
@@ -33,17 +33,16 @@ stdenv.mkDerivation {
...
-        --replace /usr/bin/udevadm ${systemd}/bin/udevadm
-
       sed -i /DEFAULT_SYS_DIR/d Makefile.in
       sed -i /DEFAULT_PROFILE_DIR/d conf/Makefile.in
+    '' + stdenv.lib.optionalString (systemd != null) ''
+      substituteInPlace scripts/lvm2_activation_generator_systemd_red_hat.c \
+        --replace /usr/bin/udevadm ${systemd}/bin/udevadm
     '';
 
   # gcc: error: ../../device_mapper/libdevice-mapper.a: No such file or directory
```
Comment (joachifm) [inline pkgs/os-specific/linux/lvm2/default.nix]: Oops, forgot to remove this comment ...

**Comment by joachifm** [comment]
@GrahamcOfBorg build systemd_with_lvm2

**Review by 7c6f434c — APPROVED** [review]

**Comment by kirelagin** [comment]
FTR I am getting the `libdevice-mapper.a` thing from https://github.com/NixOS/nixpkgs/pull/62660/commits/494d2deebfc3bde363ca5fe0bb32ddf81a86e62d.

**Comment by nh2** [comment]
@joachifm How do you use this / for what? (I'm looking into building `pkgsStatic.parted` which depends on `lvm2` so just checking if you already did something like that.)

**Comment by joachifm** [comment]
@nh2 I used it for a minimal Linux appliance that needed LVM2 but not systemd et al, so this was really part of a larger patch set that sought to allow stuff to be built without "unnecessary" (for my puroses) deps.  A version that supports static linking sounds great.

**Pull request merged**